src/Backend/Lua/Builtin.hs

```diff
@@ -215,5 +215,60 @@ builtins =
           return { x, __tag = "ShowType" }
         end
       |] )
+  , ( tcTYPEABLE, "_Typeable", [], Just (1, \[x] -> (mempty, [[lua| %x() |]]))
+    , [luaStmts|
+        local function _Typeable(x)
+          return x()
+        end
+      |] )
+  , ( tcTypeableApp, "_Typeable_app", [], Nothing
+    , [luaStmts|
+        local function _Typeable_app(pair)
+          local ta, tb = pair._1[1], pair._2[1]
+          local finger
+          if ta.fingerprint > 0 or tb.fingerprint > 0 then
+            finger = (ta.fingerprint + 10^(math.floor(math.log(ta.fingerprint))) * tb.fingerprint)
+                        / 10^(math.floor(math.log(math.abs(ta.fingerprint)) + math.log(math.abs(tb.fingerprint))))
+          else
+            finger = ta.fingerprint + 10^(math.floor(math.log(-ta.fingerprint))) * tb.fingerprint
+          end
+
+          return { { fingerprint = finger, name = ta.name .. " :$ " .. tb.name } , __tag = "TypeRep" }
+        end
+      |] )
+
+-- Type _Typeable_app instance makes a decimal of the concatenation of
```

Comment:
Hrmr, can you explain? Is this a backend issue, is there anything I can do to fix it?